src/components/TodoModal/TodoModal.tsx

```diff
@@ -15,25 +29,33 @@
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              Todo #{chosenTodo.id}
             </div>
 
             {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-            <button type="button" className="delete" data-cy="modal-close" />
+            <button
+              onClick={() => onChoosingTodo(null)}
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+            />
```

Comment:
You are interacting with the DOM directly by using the button's onClick method. It's recommended to use React as much as possible instead of interacting with the DOM directly. Consider using the synthetic event system provided by React.